### ISSUE TITLE
Customize slack notifications

### ISSUE THREAD

**Issue by glatapoui** [firehol/netdata#2775]
*TL;DR*: I would like to configure the icon, the message and the attachments of a slack notification.

In our setup, we have multiple production servers each with a tailored installation of netdata (thanks to Ansible) and they are all sending alerts to our slack. 
We have one channel per environment (prod/staging/dev), and that means multiple servers are sending to the same channel. And sometimes, it gets difficult to identify which server was impacted by an alert. So instead of displaying the netdata icon, it would help if I could use a custom icon. 

Then, I would like to use a specific image for different kinds of failures (disk, CPU, memory, Apache,...) instead of the big check or the warnings images.

Finally, regarding the timestamp footer, I do not totally understand the difference we can sometimes see between that one and the one when the message was published. Sometimes it's the same, sometime there is a difference in minutes and sometimes it's an hour. Is that the time at which the failure began ?   I would have liked to see that time on the time zone of the server, but if I read Slack's API properly, it's not really possible... What I would like to do would be to add the output of a command (like `date`) 
to that footer.

Thank you for your patience :-)
Joël.

**Comment by ktsaou** [comment]
## Custom icons

Use [this](https://github.com/firehol/netdata/blob/87aa54acd05c6814d775b56ca7326828664a19cf/conf.d/health_alarm_notify.conf#L40). This allows you to have a different setting for your production, development and staging environments, so different icons for each.

## customizing icons per type of alarm

hm... this could be done based on the chart name?
how do you plan to do it?

## time differences

The time at the bottom of the alarm, is the time the alarm generated, without any hysteresis applied.

You can customize this footer [here](https://github.com/firehol/netdata/blob/87aa54acd05c6814d775b56ca7326828664a19cf/plugins.d/alarm-notify.sh#L1141). So, copy `alarm-notify.sh` to `my-alarm-notify.sh` and change also this in `netdata.conf`:

```
[health]
	script to execute on alarm = /usr/libexec/netdata/plugins.d/alarm-notify.sh
```

Keep in mind though, that if you change `alarm-notify.sh`, your code will need to be synced if the command line parameters of `alarm-notify.sh` change in the future.

Another way, is to [use this feature](https://github.com/firehol/netdata/blob/87aa54acd05c6814d775b56ca7326828664a19cf/conf.d/health_alarm_notify.conf#L322-L381) to supply your own `curl` command for sending slack notifications. This way, you will continue to use the netdata shipped `alarm-notify.sh` and you will be able to send custom slack messages.

**Comment by glatapoui** [comment]
Thank you very much @ktsaou for your quick and detailed answer, I will try those settings with my next devops iteration and keep you posted with how it went. 

More generally, thank you for your support and the work you put in this tool. It really is a wonderful piece of software. Kudos !

**Comment by ktsaou** [comment]
ok. I am closing this, but you still post when ready.
